File: frontend/src/types/orgUnit.ts
export type OrgUnitStatus = 'TRIAL' | 'ACTIVE' | 'INACTIVE' | 'ARCHIVED'

export interface OrgUnitTreeResponse {
  id: string
  name: string
  code: string
  parentId: string | null
  type: string
  path: string
  level: number
  status: string
  logoUrl: string | null
  memberCount?: number
  allowedRoles?: { id: string; name: string; isSystem: boolean }[]
  assignedRoles?: { id: string; name: string; isSystem: boolean }[]
  children: OrgUnitTreeResponse[]
}

export interface OrgUnitResponse {
  id: string
  name: string
  code: string
  parentId: string | null
  orgHierarchyId: string
  organizationId: string
  type: string
  path: string
  level: number
  email: string | null
  phone: string | null
  address: string | null
  provinceId: string | null
  provinceName: string | null
  districtId: string | null
  districtName: string | null
  status: string
  logoUrl: string | null
  memberCount?: number
  allowedRoles?: { id: string; name: string; isSystem: boolean }[]
  assignedRoles?: { id: string; name: string; isSystem: boolean }[]
  createdAt: string
  updatedAt: string
}

export interface OrgHierarchyLevelResponse {
  id: string
  unitTypeName: string
  managerRoleLabel: string | null
  levelOrder: number
  roleLevel: number
}

export interface CreateOrgUnitRequest {
  name: string
  code: string
  parentId?: string
  orgHierarchyId: string
  email?: string
  phone?: string
  address?: string
  provinceId?: number
  districtId?: number
}

export interface UpdateOrgUnitRequest {
  name?: string
  code?: string
  orgHierarchyId?: string
  email?: string
  phone?: string
  address?: string
  provinceId?: number
  districtId?: number
  status?: OrgUnitStatus
}

export interface MoveOrgUnitRequest {
  newParentId: string | null
}

File: backend/src/main/java/com/kpitracking/entity/OrgUnit.java
package com.kpitracking.entity;

import com.kpitracking.enums.OrgUnitStatus;
import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.SQLRestriction;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import java.time.Instant;
import java.util.ArrayList;
import java.util.List;
import java.util.UUID;

@Entity
@Table(name = "org_units")
@EntityListeners(AuditingEntityListener.class)
@SQLRestriction("deleted_at IS NULL")
@Getter @Setter @Builder @NoArgsConstructor @AllArgsConstructor
public class OrgUnit {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @Column(name = "name", nullable = false)
    private String name;

    @Column(name = "code", unique = true)
    private String code;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "parent_id")
    private OrgUnit parent;

    @OneToMany(mappedBy = "parent", fetch = FetchType.LAZY)
    @Builder.Default
    private List<OrgUnit> children = new ArrayList<>();

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "org_hierarchy_id", nullable = false)
    private OrgHierarchyLevel orgHierarchyLevel;

    @Column(name = "path", nullable = false)
    private String path;

    @Column(name = "email")
    private String email;

    @Column(name = "phone")
    private String phone;

    @Column(name = "address")
    private String address;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "province_id")
    private Province province;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "district_id")
    private District district;

    @Column(name = "logo_url")
    private String logoUrl;

    @Enumerated(EnumType.STRING)
    @Column(name = "status", nullable = false)
    @Builder.Default
    private OrgUnitStatus status = OrgUnitStatus.ACTIVE;

    @CreatedDate
    @Column(name = "created_at", updatable = false)
    private Instant createdAt;

    @LastModifiedDate
    @Column(name = "updated_at")
    private Instant updatedAt;

    @Column(name = "deleted_at")
    private Instant deletedAt;

    @ManyToMany
    @JoinTable(
        name = "role_scopes",
        joinColumns = @JoinColumn(name = "org_unit_id"),
        inverseJoinColumns = @JoinColumn(name = "role_id")
    )
    @Builder.Default
    private List<Role> allowedRoles = new ArrayList<>();
}
